File: BOJ/BOJ - 9465.py
"""
author : Kwak Daeun
github : https://github.com/daeunkwak

"""
T = int(input())
for _ in range (T) :
    st = []
    n = int(input())
    st.append(list(map(int, input().split())))
    st.append(list(map(int, input().split())))

    st[0][1] = st[0][1] + st[1][0]
    st[1][1] = st[1][1] + st[0][0]

    for i in range(2, n) :
        st[0][i] = st[0][i] + max(st[1][i - 1], st[1][i - 2])
        st[1][i] = st[1][i] + max(st[0][i - 1], st[0][i - 2])

    print(max(st[0][n - 1], st[1][n - 1]))

File: BOJ/BOJ - 9465.java
/*
 author : https://github.com/daeunkwak/Problem-Solving
 title : 스티커
 description : 다이나믹 프로그래밍
 date : 2022-07-20
 */

package DP;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class BOJ_9465 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine());

        for(int i = 0; i < T; i++){
            int n = Integer.parseInt(br.readLine());
            int[][] sticker = new int[2][n + 1];
            int[][] dp = new int[2][n + 1];

            // 스티커값 입력
            for(int k = 0; k < 2; k++) {
                StringTokenizer st = new StringTokenizer(br.readLine());
                for (int j = 1; j <= n; j++) {
                    sticker[k][j] = Integer.parseInt(st.nextToken());
                }
            }

            dp[0][1] = sticker[0][1];
            dp[1][1] = sticker[1][1];
            for(int j = 2; j <= n; j++){
                dp[0][j] = Math.max(dp[1][j - 1], dp[1][j - 2]) + sticker[0][j];
                dp[1][j] = Math.max(dp[0][j - 1], dp[0][j - 2]) + sticker[1][j];
            }
            System.out.println(Math.max(dp[0][n], dp[1][n]));
        }
    }
}
